src/Loggers/Loggers.cpp

```diff
@@ -53,215 +65,213 @@ static std::string renderFileNameTemplate(time_t now, const std::string & file_p
     return path.replace_filename(ss.str());
 }
 
+size_t getLogFileMaxSize(const std::string & max_size_str)
+{
+    if (max_size_str == "never")
+        return 0;
+
+    std::string_view max_size_view{max_size_str};
+    size_t multiplier = 1;
+    if (max_size_view.ends_with('K'))
+    {
+        max_size_view.remove_suffix(1);
+        multiplier = 1_KiB;
+    }
+    else if (max_size_view.ends_with('M'))
+    {
+        max_size_view.remove_suffix(1);
+        multiplier = 1_MiB;
+    }
+    else if (max_size_view.ends_with('G'))
+    {
+        max_size_view.remove_suffix(1);
+        multiplier = 1_GiB;
+    }
+
+    return DB::parse<size_t>(max_size_view) * multiplier;
+}
+
+}
+
 /// NOLINTBEGIN(readability-static-accessed-through-instance)
 
-void Loggers::buildLoggers(Poco::Util::AbstractConfiguration & config, Poco::Logger & logger /*_root*/, const std::string & cmd_name)
+void Loggers::buildLoggers(Poco::Util::AbstractConfiguration & config, const std::string & /*cmd_name*/, bool allow_console_only)
 {
+    /// if we are remapping executable, we can start another thread only AFTER we are done
+    /// with remapping
+    if (!config.getBool("remap_executable", false))
+        DB::startQuillBackend();
+
+    std::vector<std::shared_ptr<quill::Sink>> sinks;
+
     auto current_logger = config.getString("logger", "");
-    if (config_logger.has_value() && *config_logger == current_logger)
+    if (config_logger.has_value())// && *config_logger == current_logger)
```

Comment:
This seems odd and maybe a leftover. Does this mean we can't change the config of the logger at runtime and, for example, change the log level?

Reply:
This was a hack, so I added `initialized` flag.
We can't change loggers at runtime, we can only change levels but this is handled by function `updateLevels` below.